### PR TITLE
💄#188: Add banner component

### PR DESCRIPTION
**Pull request by xusandrew**
## Description of Changes

- banner component
- displayed on EVERY page of the portal (confirmed with Patrick)
-  option for dismissable / undismissable
- styling for info/warning/error types
- controlled by env vars*

#### To control the banners, pass in a json string of a list of banner objects to the BANNERS env var. For a sample: 
```
# BANNERS
BANNERS=[{"type":"INFO","dismissable":false,"enTitle":"English title [link](https://www.youtube.com/)","frTitle":"Francais tite","enMessage":"Click this [link](https://www.youtube.com/)","frMessage":"BONJOUR"},{"type":"WARNING","dismissable":true,"enMessage":"Warning Banner Title Would Go Here","frMessage":"BONJOUR"},{"type":"ERROR","dismissable":true,"enTitle":"English title","frTitle":"Francais tite","enMessage":"Critical Banner Title Would Go Here","frMessage":"BONJOUR"}]
```
The type for the banner object:
```
export type BannerData = {
	type: BannerType;
	dismissable: boolean;
	enTitle?: string;
	frTitle?: string;
	enMessage: string;
	frMessage: string;
};
```
#### Note that both the title and message strings support markdown, thus links/other html elements are supported. Library used for this is `react-markdown`

Any input of how to improve how I pass in info into env vars would be appreciated, I can see this probably isnt the cleanest way

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

**Review by mistryrn — COMMENTED** [review]
Tested locally following your instructions and the banners look great! 🙂 

A few questions that might help inform the direction we go for the banner env vars:
1. Will the banners ever need to be non-dismissable?
2. Will there ever be more than one banner of a certain type?
3. Do the banners ever need to have a _message_ in addition to the **title** that we currently have? For example, something like:
```
Title: OHCRN will be down for maintenance from 12am EST to 6am EST
Message: We apologize for any inconvenience, please reach out to ohcrn@example.com for any urgent questions
```
4. Follow-up to 3, if there could be messages, would those messages ever need to contain HTML (for example, links)?

Wanted to ask here first, in case you've already got the answers to these. But if not we can bring it up during the F2F and ask Patrick and Brandon

**Comment by xusandrew** [comment]
banners CAN be non-dismissible (check with ARGO)

**Comment by xusandrew** [comment]
YES to message but need new mocks

**Comment by xusandrew** [comment]
YES to html/links in messages

**Comment by xusandrew** [comment]
made changes and updated pr description!

File: apps/consent-ui/src/config/appConfig.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 import 'server-only';
 
-import { defaultAppConfig, AppConfig } from 'src/config/types';
+import { defaultAppConfig, AppConfig } from './types';
```
Comment (mistryrn) [inline apps/consent-ui/src/config/appConfig.ts:22]: Let's revert this back to the full path just for consistency with the clientConfig:
```suggestion
import { defaultAppConfig, AppConfig } from 'src/config/types';
```

**Review by mistryrn — COMMENTED** [review]
Looking good! Just one small change request, plus the investigation into the sidebar layout that we chatted about earlier 🙂

**Comment by xusandrew** [comment]
@mistryrn addressed ur comment + fixed the styles for the sidebarlayout!

**Comment by mistryrn** [comment]
@xusandrew thanks for the quick fixes! Just one last styling bug – seems like the Clinician Invite page sidebar layout is still a bit wonky for a certain breakpoint – see the dimensions in this screen recording:

https://github.com/OHCRN/platform/assets/3392776/c90c0a0a-7df9-4625-ae28-cfa96492e2cc

**Review by mistryrn — APPROVED** [review]
Looks great! Thanks for fixing all those small styling bugs so quickly 😄

**Pull request merged**